src/overlays/actors/ovl_En_Box/z_en_box.c

```diff
@@ -28,66 +68,651 @@ const ActorInit En_Box_InitVars = {
+            this->movementFlags &= ~ENBOX_MOVE_FALL_ANGLE_SIDE;
+        } else {
+            this->movementFlags |= ENBOX_MOVE_FALL_ANGLE_SIDE;
+        }
+        if (this->type == ENBOX_TYPE_BIG_SWITCH_FLAG_FALL) {
+            this->dyna.actor.velocity.y = -this->dyna.actor.velocity.y * 0.55f;
+        } else {
+            this->dyna.actor.velocity.y = -this->dyna.actor.velocity.y * 0.65f;
+        }
+        if (this->dyna.actor.velocity.y < 5.5f) {
+            this->dyna.actor.shape.rot.z = 0;
+            this->dyna.actor.world.pos.y = this->dyna.actor.floorHeight;
+            EnBox_SetupAction(this, EnBox_WaitOpen);
+        }
+        Audio_PlaySfxAtPos(&this->dyna.actor.projectedPos, NA_SE_EV_TRE_BOX_BOUND);
+        EnBox_SpawnDust(this, globalCtx);
+    }
+    yDiff = this->dyna.actor.world.pos.y - this->dyna.actor.floorHeight;
+    if (this->movementFlags & ENBOX_MOVE_FALL_ANGLE_SIDE) {
+        this->dyna.actor.shape.rot.z = yDiff * 50.0f;
+    } else {
+        this->dyna.actor.shape.rot.z = -yDiff * 50.0f;
+    }
+}
+
+void EnBox_FallOnSwitchFlag(EnBox* this, GlobalContext* globalCtx) {
+    func_800B8C50(&this->dyna.actor, globalCtx);
+    if (this->unk_1A0 >= 0) {
+        EnBox_SetupAction(this, EnBox_Fall);
+        func_800C6314(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+    } else if (this->unk_1A0 >= -11) {
+        this->unk_1A0++;
+    } else if (Flags_GetSwitch(globalCtx, this->switchFlag)) {
+        this->unk_1A0++;
+    }
+}
+
+void EnBox_AppearSwitchFlag(EnBox* this, GlobalContext* globalCtx) {
+    func_800B8C50(&this->dyna.actor, globalCtx);
+    if (Flags_GetSwitch(globalCtx, this->switchFlag)) {
+        if (ActorCutscene_GetCanPlayNext(this->cutsceneIdxA)) {
+            ActorCutscene_StartAndSetUnkLinkFields(this->cutsceneIdxA, &this->dyna.actor);
+            EnBox_SetupAction(this, func_80868AFC);
+            this->unk_1A0 = -30;
+        } else {
+            ActorCutscene_SetIntentToPlay(this->cutsceneIdxA);
+        }
+    }
+}
+
+void EnBox_AppearOnRoomClear(EnBox* this, GlobalContext* globalCtx) {
+    func_800B8C50(&this->dyna.actor, globalCtx);
+    if (Flags_GetClearTemp(globalCtx, this->dyna.actor.room)) {
+        if (ActorCutscene_GetCanPlayNext(this->cutsceneIdxA)) {
+            ActorCutscene_StartAndSetUnkLinkFields(this->cutsceneIdxA, &this->dyna.actor);
+            Flags_SetClear(globalCtx, this->dyna.actor.room);
+            EnBox_SetupAction(this, func_80868AFC);
+            this->unk_1A0 = -30;
+        } else {
+            ActorCutscene_SetIntentToPlay(this->cutsceneIdxA);
+        }
+    }
+}
+
+void func_80868AFC(EnBox* this, GlobalContext* globalCtx) {
+    if (func_800F22C4(this->cutsceneIdxA, &this->dyna.actor) != 0 || this->unk_1A0 != 0) {
+        EnBox_SetupAction(this, func_80868B74);
+        this->unk_1A0 = 0;
+        func_80867FBC(&this->unk_1F4, globalCtx, (this->movementFlags & ENBOX_MOVE_0x80) != 0);
```

Comment:
Is the `!= 0` required to match?
```suggestion
        func_80867FBC(&this->unk_1F4, globalCtx, this->movementFlags & ENBOX_MOVE_0x80);
```

Reply:
that's non-equivalent. since this is a function argument and not an if statement, != 0 forces the value to be 0,1 instead of 0, 0x80